enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,435 @@
+| AllowedUnsafeSysctls            | sysctls                     | No impact--the handling of sysctls should not change if a pod is in a user namespace.                                                                                                                                     |
+| ForbiddenSysctls                | sysctls                     | No impact--the handling of sysctls should not change if a pod is in a user namespace.                                                                                                                                     |
+
+##### `UserNamespaceLevel` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // userNamespaceLevel determines if the policy allows host users in containers.
+       // Valid values are "AllowHostLevel", "RequirePodLevel", and omitted.
+       // When "AllowHostLevel" is set, a pod author may set `hostUsers` to either `true` or `false`.
+       // When "RequirePodLevel" is set, a pod author must `hostUsers` to `false`.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default,
+       // which is subject to change over time.
+       // The current default is "AllowHostLevel".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityStandards
+       // +kubebuilder:validation:Enum="AllowHostLevel";"RequirePodLevel";""
+       // +optional
+       UserNamespaceLevel NamespaceLevelType `json:"userNamespaceLevel" protobuf:"bytes,26,opt,name=userNamespaceLevel"`
+...
+// NamespaceLevelType shows the allowable values for the UserNamespaceLevel field.
+type NamespaceLevelType string
+...
+const (
+       // NamespaceLevelAllowHost allows a pod to set `hostUsers` field to either `true` or `false`
+       NamespaceLevelAllowHost NamespaceLevelType = "AllowHostLevel"
+       // NamespaceLevelRequirePod requires the `hostUsers` field be `false` in a pod.
+       NamespaceLevelRequirePod NamespaceLevelType = "RequirePodLevel"
+       // NamespaceLevelEmpty will set to the default, which is currently "AllowHostLevel".
+       NamespaceLevelEmpty NamespaceLevelType = ""
+
+...
+)
+
+```
+
+This field branches from its fellow `AllowHost*` namespace fields to follow current API conventions, which maintain boolean fields are not extendable enough.
+
+##### restricted-v3
+
+This SCC profile will be identical to the existing `restricted-v2`, except it will set the `UserNamespaceLevel` to `RequirePodLevel`,
+thus forcing pods to be in a user namespace. This will make it a more restrictive profile, as the user on the host will not be
+the same as the one inside the container.
+
+Note: upstream Kubernetes deemed `runAsUser` to be a safe field to use for pods with `hostUsers: false`. However, the philosophy
+of SCC utilizes defense-in-depth. `restricted-v3` will be the most restrictive policy.
+
+##### nonroot-v3
```

Comment:
Does this map to PSA restricted in the PSA auto-syncer?

Reply:
I don't know what the auto-syncer is. A combination of `nonroot-v3` and `restricted-v2` are equivalent to PSA restricted (if no user namespace, restrict runAsUser, if yes user namespace, allow any runAsUser)

Reply:
the PSA autosyncer looks at an SCC and determines the tightest PSA level that allows any pod allowed by the SCC to run in the namespace.

If all pods allowed by nonroot-v3 can run in the restricted namespace do you anticipate allowing all authenticated users to use nonroot-v3 by default?

Reply:
I would say for the very very security minded folks, restricted-v3 should be the best and may need to be required for some users, but nonroot-v3 is also very safe and I'd expect the majority of folks to be able to use it for what they need

Reply:
reduced just to `restricted-v3` for now

Reply:
@deads2k,

if the feature gate `UserNamespacesPodSecurityStandards` is enabled, then nonroot-v3 maps to pod security standard's new restricted, ref: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#integration-with-pod-security-admission-checks.

This is why I wanted to have it in. `restricted-v3` is more restrictive then restricted with the feature gate mentioned above.

Reply:
The decision to push nonroot-v3 out of the initial release doesn't preclude it coming in later, but it raised concerns about downgrade and default access.  It's a good candidate to bring back either prior to or after GA of the feature overall.